src/main/java/sg/ncl/MainController.java

```diff
@@ -3274,49 +3275,100 @@ public String adminUsersManagement(Model model, HttpSession session) {
 
     @RequestMapping("/admin/usage")
     public String adminTeamUsage(Model model,
-                                 @RequestParam(value = "team", required = false) String team,
                                  @RequestParam(value = "start", required = false) String start,
                                  @RequestParam(value = "end", required = false) String end,
-                                 HttpSession session) {
+                                 @RequestParam(value = "organizationType", required = false) String organizationType,
+                                 @RequestParam(value = "team", required = false) String team,
+                                 final RedirectAttributes redirectAttributes,
+                                 HttpSession session) throws IOException, WebServiceRuntimeException {
         if (!validateIfAdmin(session)) {
             return NO_PERMISSION_PAGE;
         }
+
         DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
-        ZonedDateTime now = ZonedDateTime.now();
+        ZonedDateTime nowDate = ZonedDateTime.now();
+        String now = nowDate.format(formatter);
         if (start == null) {
-            ZonedDateTime startDate = now.with(firstDayOfMonth());
+            ZonedDateTime startDate = nowDate.with(firstDayOfMonth());
+            if (nowDate.getDayOfMonth() == 1) {
+                startDate = startDate.minusMonths(1);
+            }
             start = startDate.format(formatter);
         }
         if (end == null) {
-            ZonedDateTime endDate = now.with(lastDayOfMonth());
+            ZonedDateTime endDate = nowDate.minusDays(1);
             end = endDate.format(formatter);
         }
+        if (now.compareTo(start) <= 0 || now.compareTo(end) <= 0) {
+            redirectAttributes.addFlashAttribute(MESSAGE, "Period selected is on or beyond current date (today).");
+            return REDIRECT_TEAM_USAGE;
+        }
 
         // get list of teamids
         HttpEntity<String> request = createHttpEntityHeaderOnly();
         ResponseEntity responseEntity = restTemplate.exchange(properties.getSioTeamsUrl(), HttpMethod.GET, request, String.class);
         JSONArray jsonArray = new JSONArray(responseEntity.getBody().toString());
 
+        List<Team2> searchTeams = new ArrayList<>();
         TeamManager2 teamManager2 = new TeamManager2();
-        for (int i = 0; i < jsonArray.length(); i++) {
-            JSONObject jsonObject = jsonArray.getJSONObject(i);
-            Team2 one = extractTeamInfo(jsonObject.toString());
-            teamManager2.addTeamToTeamMap(one);
-        }
+        getSearchTeams(organizationType, team, jsonArray, searchTeams, teamManager2);
+
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)